lib/src/config_provider/config_spec.dart

```diff
@@ -591,7 +591,7 @@ class StringConfigSpec<RE extends Object?> extends ConfigSpec<String, RE> {
     if (!o.checkType<String>(log: log)) {
       return false;
     }
-    if (_regexp != null && !_regexp.hasMatch(o.value as String)) {
+    if (!(_regexp?.hasMatch(o.value as String) ?? true)) {
```

Comment:
nit: does the `!` bind stronger than the `??`, I guess it does. Maybe add parenthesis.